angular/index.d.ts

```diff
@@ -901,8 +901,17 @@ declare namespace angular {
      */
     interface ILocationService {
         absUrl(): string;
+
+        /**
+         * Returns the hash fragment
+         */
         hash(): string;
-        hash(newHash: string): ILocationService;
+
+        /**
+         * Changes the hash fragment and returns `$location`
+         */
+        hash(newHash: string|null): ILocationService;
```

Comment:
This function is a getter-setter. When invoked without parameters, it's a getter.